build.xml

```
@@ -390,6 +390,62 @@
         </antcall>
     </target>
 
+    <target name="compare-versions" description="Compares the canonicalized sources of the MEI dev version with the previous stable version and creates an HTML output; custom versions and output folder can be set via -Dsource, -Dold and -Doutput input params." depends="canonicalize-source">
+        <!-- set source file -->
+        <property name="source.default" value="${dir.build}/mei-source_canonicalized.xml"/>
+        <echo level="verbose">source.default: ${source.default}</echo>
+        <echo level="verbose">source user input: ${source}</echo>
+
+        <condition property="source.file" value="${source}" else="${source.default}">
+            <isset property="source"/>
+        </condition>
+        <echo>Using source.file: ${source.file}</echo>
+
+        <!-- set old file to compare with -->
+        <property name="old.default" value="${dir.build}/mei-source_canonicalized_v4.0.1.xml"/>
+        <condition property="old.default-available" value="true">
+                <available file="${old.default}"/>
+        </condition>
+        <get src="https://raw.githubusercontent.com/music-encoding/schema/main/4.0.1/mei-source_canonicalized_v4.0.1.xml" dest="${old.default}" unless:true="${old.default-available}"/>
+
+        <echo level="verbose">old.default: ${old.default}</echo>
+        <echo level="verbose">old user input: ${old}</echo>
+
+        <condition property="old.file" value="${old}" else="${old.default}">
+            <isset property="old"/>
+        </condition>
+        <echo>Using old.file: ${old.file}</echo>
+
+        <!-- check if old.file is available, otherwise fail early -->
+        <condition property="old.available" value="true" else="false">
+            <available file="${old.file}"/>
+        </condition>
+        <echo message="old.file available: ${old.available}"></echo>
+
+        <fail unless:true="${old.available}">Unfortunately, the $old.file ${old.file} is not available at the specified location.</fail>
+
+        <!-- set ouput folder -->
+        <property name="output.default" value="./utils/compare_versions/"/>
+        <echo level="verbose">output.default: ${output.default}</echo>
+        <echo level="verbose">output user input: ${output}</echo>
+
+        <condition property="output.folder" value="${output}" else="${output.default}">
+            <isset property="output"/>
+        </condition>
+        <echo>Using output folder: ${output.folder}</echo>
+
+        <!-- compare versions -->
+        <java classname="net.sf.saxon.Transform" classpathref="mei.classpath" failonerror="true" fork="true">
+            <arg value="-s:${source.file}"/>
+            <arg value="-xsl:${basedir}/utils/compare_versions/comparison.xsl"/>
+            <arg value="-xi:on"/>
+            <arg value="output.folder=${output.folder}"/>
+            <arg value="old.version.filename=/${old.file}"/>
```

Of course you're right that an absolute path will make the transformation work, no matter, where `${old.source}` is located. Running `ant compare-versions` natively from the music-encoding folder fails though. Moreover, setting it absolute here will prevent submitting an arbitrary file path. 

```suggestion
            <arg value="old.version.filename=${old.file}"/>
```

The absolute path for `${old.default}` could be achieved by prepending `${basedir}` in its definition (see above).

---

```suggestion
            <arg value="old.version.filename=file:${old.file}"/>
```